salesforce/client.bal

```diff
@@ -245,6 +245,73 @@ public isolated client class Client {
         return check self.salesforceClient->delete(path);
     }
 
+    # Lists reports.
+    #
+    # + return - Reports if successful or else `error`
+    @display {label: "List Reports"}
+    isolated remote function listReports() returns @display {label: "Reports"} Report[]|error {
+        string path = utils:prepareUrl([API_BASE_PATH, ANALYTICS, REPORTS]);
+        return check self.salesforceClient->get(path);
+    }
+
+    # Deletes a report.
+    #
+    # + reportId - Report Id
+    # + return - `()` if the report deletion is successful or else an error
+    @display {label: "Delete Report"}
+    isolated remote function deleteReport(@display {label: "Report Id"} string reportId) returns error? {
+        string path = utils:prepareUrl([API_BASE_PATH, ANALYTICS, REPORTS, reportId]);
+        return check self.salesforceClient->delete(path);
+    }
+
+    # Runs an instance of a report synchronously.
+    #
+    # + reportId - Report Id
+    # + return - Reports if successful or else `error`
```

Comment:
Shall we update the return when successful with the proper record returned.

Reply:
Done